### PR TITLE
TimeRangeFilter simplification

### PR DESCRIPTION
**Pull request by ubitux**

### CODE REVIEW

**Comment by mbouron** [comment]
nit: `doc: fix media ...` -> `doc: simplify media ...`

**Comment by ubitux** [comment]
> nit: `doc: fix media ...` -> `doc: simplify media ...`

It was actually broken; I added a description elaborating on that: "This code was missing a Program node."

File: pynopegl-utils/pynopegl_utils/examples/timeranges.py [hunk 1]
```diff
@@ -30,32 +30,21 @@ def parallel_playback(cfg: SceneCfg, fast=True, segment_time=2.0, constrained_ti
...
-
-        t += 2 * segment_time
-
-    rf1.add_ranges(*rr1)
-    rf2.add_ranges(*rr2)
+    i = 0
+    while end_time < cfg.duration:
+        end_time = (i + 1) * segment_time
+        rr, render = (rr2, render2) if i & 1 else (rr1, render1)
+        tseg = ngl.TimeRangeFilter(render, start=i * segment_time, end=(i + 1) * segment_time)
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/examples/timeranges.py:40]: You can use end_time here.

File: pynopegl-utils/pynopegl_utils/examples/timeranges.py [hunk 2]
```diff
@@ -30,32 +30,21 @@ def parallel_playback(cfg: SceneCfg, fast=True, segment_time=2.0, constrained_ti
...
-        rr2.append(ngl.TimeRangeModeNoop(t))
-        rr2.append(ngl.TimeRangeModeCont(t + segment_time))
-
-        t += 2 * segment_time
-
-    rf1.add_ranges(*rr1)
-    rf2.add_ranges(*rr2)
+    i = 0
+    while end_time < cfg.duration:
+        end_time = (i + 1) * segment_time
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/examples/timeranges.py:38]: Can you declare start_time here (to improve readability).

File: pynopegl-utils/pynopegl_utils/examples/timeranges.py [hunk 3]
```diff
@@ -30,32 +30,21 @@ def parallel_playback(cfg: SceneCfg, fast=True, segment_time=2.0, constrained_ti
...
+        rr, render = (rr2, render2) if i & 1 else (rr1, render1)
```
Comment (mbouron) [inline pynopegl-utils/pynopegl_utils/examples/timeranges.py:39]: nit: I find the new logic a bit harder to read.

File: libnopegl/src/nopegl.h.in [hunk 4]
```diff
@@ -249,9 +249,7 @@ struct ngl_node;
 #define NGL_NODE_TEXTUREVIEW            NGLI_FOURCC('T','e','x','V')
 #define NGL_NODE_TIME                   NGLI_FOURCC('T','i','m','e')
 #define NGL_NODE_TIMERANGEFILTER        NGLI_FOURCC('T','R','F','l')
-#define NGL_NODE_TIMERANGEMODECONT      NGLI_FOURCC('T','R','M','C')
-#define NGL_NODE_TIMERANGEMODENOOP      NGLI_FOURCC('T','R','M','N')
-#define NGL_NODE_TIMERANGEMODEONCE      NGLI_FOURCC('T','R','M','1')
+#define NGL_NODE_TIMESEGMENT            NGLI_FOURCC('T','S','e','g')
```
Comment (mbouron) [inline libnopegl/src/nopegl.h.in:252]: I guess this is a leftover from a previous iteration ?

File: libnopegl/src/node_timerangefilter.c [hunk 5]
```diff
@@ -30,107 +30,52 @@
...
+    double end_time;
+    double render_time;
     double prefetch_time;
-    double max_idle_time;
 };
 
 struct timerangefilter_priv {
-    int current_range;
     int drawme;
+    int updated;
```
Comment (mbouron) [inline libnopegl/src/node_timerangefilter.c:41]: nit: maybe declare it before drawme (updates happen before draw).

**Review by mbouron — CHANGES_REQUESTED** [review]
Other than my remarks, LGTM.

File: pynopegl-utils/pynopegl_utils/examples/timeranges.py (hunk 1 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/examples/timeranges.py]: Ah yeah my bad

File: pynopegl-utils/pynopegl_utils/examples/timeranges.py (hunk 2 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/examples/timeranges.py:40]: Indeed, done

File: pynopegl-utils/pynopegl_utils/examples/timeranges.py (hunk 3 above)
Comment (ubitux) [reply pynopegl-utils/pynopegl_utils/examples/timeranges.py]: I added parenthesis around the destination tuple and with your suggestions the code looks sane to me:

```python
    end_time = 0.0
    rr1 = []
    rr2 = []
    i = 0
    while end_time < cfg.duration:
        (rr, render) = (rr2, render2) if i & 1 else (rr1, render1)
        start_time = i * segment_time
        end_time = (i + 1) * segment_time
        tseg = ngl.TimeRangeFilter(render, start_time, end_time)
        if constrained_timeranges:
            tseg.set_prefetch_time(segment_time / 3)
        rr.append(tseg)
        i += 1
```

File: libnopegl/src/nopegl.h.in (hunk 4 above)
Comment (ubitux) [reply libnopegl/src/nopegl.h.in]: Oops yeah, removed

File: libnopegl/src/node_timerangefilter.c (hunk 5 above)
Comment (ubitux) [reply libnopegl/src/node_timerangefilter.c:40]: Sure, done

**Review by mbouron — APPROVED** [review]

**Pull request merged**